[webkitpy] Update cryptography and openssl dependencies

JonWBedard opened the pull request:
#### b8390e24dbe69edce19760bc41b1a654d9dedc9b
<pre>
[webkitpy] Update cryptography and openssl dependencies
<a href="https://bugs.webkit.org/show_bug.cgi?id=268118">https://bugs.webkit.org/show_bug.cgi?id=268118</a>
<a href="https://rdar.apple.com/121462885">rdar://121462885</a>

Reviewed by Sam Sneddon and BJ Burg.

We need to update boto for testing. This requires updating cryptography,
which requires updating OpenSSL. Since cryptography is coupled to OpenSSL,
we should declare their dependencies in the same place.

* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py: Update boto and dependencies.
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py: Update cryptography on
newer version of Python, pair OpenSSL with cryptography definitions.
* Tools/Scripts/webkitpy/autoinstalled/twisted.py: Move OpenSSL dependency to webkitcorepy.

Canonical link: <a href="https://commits.webkit.org/277995@main">https://commits.webkit.org/277995@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2ec90f61668fa695e6238e4f3f46e4a7abe0d8ed

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48919 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28132 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51886 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51606 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44989 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51224 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34084 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25660 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39987 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/49658 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25776 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42178 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21094 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/48779 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23236 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43350 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6974 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45167 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43851 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53517 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23970 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20229 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47294 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/48948 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25233 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42387 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46248 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26042 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7048 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24953 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders reviewed (approved):
I still dislike how complicated the version matrix is (which makes it near impossible to check we're actually installing compatible versions which satisfy each other's requirements), but I guess the one thing touched here that we *don't* have test coverage of (between `webkit-misc` having 2.7 and 3.7, the rest of the EWS bots having 3.9) is the 3.11 branch, but you haven't really touched that, so… I guess this should be fine?

JonWBedard committed:
[webkitpy] Update cryptography and openssl dependencies
https://bugs.webkit.org/show_bug.cgi?id=268118
rdar://121462885

Reviewed by Sam Sneddon and BJ Burg.

We need to update boto for testing. This requires updating cryptography,
which requires updating OpenSSL. Since cryptography is coupled to OpenSSL,
we should declare their dependencies in the same place.

* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py: Update boto and dependencies.
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py: Update cryptography on
newer version of Python, pair OpenSSL with cryptography definitions.
* Tools/Scripts/webkitpy/autoinstalled/twisted.py: Move OpenSSL dependency to webkitcorepy.

Canonical link: https://commits.webkit.org/277995@main
